### PR TITLE
ci: trigger trivy-action after release

### PR DESCRIPTION
**Pull request by nikpivkin**
## Related issues
- Close https://github.com/aquasecurity/trivy/issues/4829

## Related PRs
- [x] https://github.com/aquasecurity/trivy-action/pull/245

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -55,3 +55,14 @@ jobs:
 
       - name: Create deb repository
         run: ci/deploy-deb.sh
+
+      - id: get_version
+        uses: battila7/get-version-action@v2
+
+      - name: Trigger ${{ github.repository_owner }}/trivy-action
```
Comment (knqyf263) [inline .github/workflows/release.yaml:62]: I'd use the official repository as much as possible. Can we use `actions/github-script` instead?
https://github.com/orgs/community/discussions/26323#discussioncomment-3251451

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -55,3 +55,14 @@ jobs:
 
       - name: Create deb repository
         run: ci/deploy-deb.sh
+
+      - id: get_version
+        uses: battila7/get-version-action@v2
```
Comment (knqyf263) [inline .github/workflows/release.yaml:60]: ditto.
https://stackoverflow.com/questions/71630227/github-action-how-to-remove-starting-v-from-variable

File: .github/workflows/release.yaml (hunk 1 above)
Comment (nikpivkin) [reply .github/workflows/release.yaml:59]: done

**Comment by nikpivkin** [comment]
@knqyf263 should i use the `ORG_REPO_TOKEN` secret to get the `PAT` token?  we need a token with the `repo` permission to run workflow in another repo.

**Comment by knqyf263** [comment]
>@knqyf263 should i use the ORG_REPO_TOKEN secret to get the PAT token? we need a token with the repo permission to run workflow in another repo.

Yes, ORG_REPO_TOKEN should have the `repo` permission.

**Comment by simar7** [comment]
https://github.com/aquasecurity/trivy-action/pull/245 was merged so we should be ready to go with this.

**Comment by knqyf263** [comment]
LGTM. @nikpivkin Did you test this workflow in your fork?

**Comment by nikpivkin** [comment]
@knqyf263 I tested the trigger logic on mock repositories

**Review by simar7 — APPROVED** [review]

**Comment by knqyf263** [comment]
It's better to use the fork repositories. Trivy is a large repository, and it is hard to mock. We could see unexpected behavior even if it works in the mock repositories.

**Comment by simar7** [comment]
> It's better to use the fork repositories. Trivy is a large repository, and it is hard to mock. We could see unexpected behavior even if it works in the mock repositories.

@nikpivkin any update on this?

**Comment by nikpivkin** [comment]
@simar7 I'm working on it

**Comment by nikpivkin** [comment]
@knqyf263 I was thinking how about moving the trigger step into a separate job that depends on the `release` job?

**Comment by simar7** [comment]
> It's better to use the fork repositories. Trivy is a large repository, and it is hard to mock. We could see unexpected behavior even if it works in the mock repositories.

@knqyf263 but on a fork the `ORG_REPO_TOKEN` for aquasec org will not be available, so the job will fail. Maybe we can try in a different repo within the `aquasecurity` organization?

**Comment by knqyf263** [comment]
>@knqyf263 I was thinking how about moving the trigger step into a separate job that depends on the release job?

Sounds good

**Comment by knqyf263** [comment]
>but on a fork the ORG_REPO_TOKEN for aquasec org will not be available, so the job will fail. Maybe we can try in a different repo within the aquasecurity organization?

You can create `simar7/trivy` and `simar7/trivy-action` then use your PAT for `ORG_REPO_TOKEN`.

**Comment by knqyf263** [comment]
Yeah, the permission is always problematic. Even I can't see that. I'll ask @itaysk.

**Comment by knqyf263** [comment]
@nikpivkin `workflow` is not checked. Do you need it?
image

**Comment by nikpivkin** [comment]
@knqyf263 Yes. This is necessary to change the Trivy version.

**Comment by knqyf263** [comment]
OK, I've added `workflow`.

**Comment by knqyf263** [comment]
@nikpivkin Please let me know if you confirm it works in your forks.

**Comment by nikpivkin** [comment]
@knqyf263 I am converting this into a draft, as it works in my forks, but it is still necessary to ensure the consistency of the database in trivy-action.

**Pull request closed without merging**